modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -71,43 +71,81 @@ export default function CartQuickAdd() {
 		fetch(productsApiURL.toString())
 			.then((response) => response.json())
 			.then((availableProducts) => {
-				setFormattedProducts(
-					availableProducts.items.map((product) => {
-						const {name, skus} = product;
-
-						return {
+				const formattedProducts = [];
+
+				availableProducts.items.forEach((product) => {
+					const {name, skus} = product;
+
+					if (product.skus.length > 1) {
+						product.skus.forEach((sku) =>
+							formattedProducts.push({
+								...sku,
+								label: name,
+								value: sku.sku,
+							})
+						);
+					}
+					else {
+						formattedProducts.push({
 							...product,
 							label: name,
 							value: skus[0].sku,
-						};
-					})
+						});
+					}
+				});
+
+				setFormattedProducts(formattedProducts);
+
+				const productsWithOptions = availableProducts.items.filter(
+					(product) => {
+						if (product.skus.length > 1) {
+							return product.skus.map((sku) => sku);
+						}
+					}
 				);
 
-				setProducts(availableProducts.items);
+				setProducts([
+					...availableProducts.items,
+					...productsWithOptions,
+				]);
 			});
 	}, [channelId]);
 
 	const handleAddToCartClick = () => {
-		const itemSKUs = selectedProducts.map((item) => item.value);
-		const readyProducts = [];
-
-		products.forEach((product) => {
-			if (
-				product.skus.length &&
-				product.skus[0] &&
-				itemSKUs.includes(product.skus[0].sku)
-			) {
+		const readyProducts = selectedProducts.map((product) => {
+			if (product.sku) {
+				const productsWithOptions = products.filter(
+					(item) => item.skus && item.skus.length > 1
+				);
+
+				const parentProduct = productsWithOptions.find((item) =>
+					item.skus.find((childSku) => childSku.sku === product.sku)
+				);
+
+				return {
+					...product,
+					name: parentProduct.name,
+					price: product.price,
+					productURLs: parentProduct.urls,
+					quantity:
+						parentProduct.productConfiguration.minOrderQuantity,
+					settings: parentProduct.productConfiguration,
+					sku: product.sku,
+					skuId: product.id,
+				};
+			}
+			else {
```

Comment:
I'm having a hard time tracking some of this logic with all the loops and searching. Would something like this make more sense?

```js
const productsWithOptions = products.filter(
	(item) => item.skus && item.skus.length > 1
);

const childSkus = productsWithOptions.map(item => item.skus);

const readyProducts = selectedProducts.map((product) => {
	if (product.sku) {
		const parentIndex = childSkus.findIndex(skus => skus.include(product.sku))

		const parentProduct = productsWithOptions[parentIndex];
		
		return {
			...product,
			name: parentProduct.name,
			price: product.price,
			productURLs: parentProduct.urls,
			quantity:
				parentProduct.productConfiguration.minOrderQuantity,
			settings: parentProduct.productConfiguration,
			sku: product.sku,
			skuId: product.id,
		};
	}
```

Reply:
@bryceosterhaus what did you imagine that happens when there is no `parentProduct`? The return object needs to look different. There are way more products without options, than those with options.

Reply:
Not sure, I was just following the code you had written, which also just passes the parentProduct to the object

Reply:
Your suggestion wasn't really working, I initially mistakenly thought you want me to replace both returns. After I course-corrected, still the `parentProduct` kept being undefined no matter what I do, but the solution I ended up with cleaned up the code, plus I verified `productsWithOptions` is valid and the `parentProduct` handling looks clean. https://github.com/liferay-frontend/liferay-portal/pull/2961/commits/cc11cffc0955100cce1806331a5c8ea7021e0ba5 Still lemme know if anything can be improved, but everything is pretty tightly wound from my experience with this component.